data-prepper-plugins/http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/HTTPSourceConfig.java

```diff
@@ -11,49 +11,40 @@
 
 package com.amazon.dataprepper.plugins.source.loghttp;
 
-import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Preconditions;
 import io.micrometer.core.instrument.util.StringUtils;
 
 import java.nio.file.Files;
 import java.nio.file.Path;
 
 public class HTTPSourceConfig {
-    static final String PORT = "port";
     static final String REQUEST_TIMEOUT = "request_timeout";
     static final String THREAD_COUNT = "thread_count";
-    static final String MAX_CONNECTION_COUNT = "max_connection_count";
     static final String MAX_PENDING_REQUESTS = "max_pending_requests";
     static final String DEFAULT_LOG_INGEST_URI = "/log/ingest";
     static final String SSL = "ssl";
     static final String SSL_CERTIFICATE_FILE = "ssl_certificate_file";
     static final String SSL_KEY_FILE = "ssl_key_file";
-    static final String SSL_KEY_PASSWORD = "ssl_key_password";
     static final int DEFAULT_PORT = 2021;
     static final int DEFAULT_REQUEST_TIMEOUT_MS = 10000;
     static final int DEFAULT_THREAD_COUNT = 200;
     static final int DEFAULT_MAX_CONNECTION_COUNT = 500;
     static final int DEFAULT_MAX_PENDING_REQUESTS = 1024;
 
-    private final int port;
-    private final int requestTimeoutInMillis;
-    private final int threadCount;
-    private final int maxConnectionCount;
-    private final int maxPendingRequests;
-    private final boolean ssl;
-    private final String sslCertificateFile;
-    private final String sslKeyFile;
-    private final String sslKeyPassword;
-
-    private HTTPSourceConfig(final int port,
-                             final int requestTimeoutInMillis,
-                             final int threadCount,
-                             final int maxConnectionCount,
-                             final int maxPendingRequests,
-                             final boolean ssl,
-                             final String sslCertificateFile,
-                             final String sslKeyFile,
-                             final String sslKeyPassword) {
+    private int port = DEFAULT_PORT;
+    @JsonProperty("request_timeout")
+    private int requestTimeoutInMillis = DEFAULT_REQUEST_TIMEOUT_MS;
+    private int threadCount = DEFAULT_THREAD_COUNT;
+    private int maxConnectionCount = DEFAULT_MAX_CONNECTION_COUNT;
```

Comment:
Why no jsonProperty annotation on top of this variable?

Reply:
The default conversion will support snake case, so this should serialize. But, I can add add it to be somewhat more thorough and consistent.